python/dask_cudf/dask_cudf/backends.py

```diff
@@ -75,19 +75,41 @@ def _nonempty_index(idx):
     raise TypeError(f"Don't know how to handle index of type {type(idx)}")
 
 
+def _nest_list_data(data, leaf_type):
+    """
+    Helper for _get_non_empty_data which creates
+    nested list data
+    """
+    data = [data]
+    while isinstance(leaf_type, cudf.ListDtype):
+        leaf_type = leaf_type.element_type
+        data = [data]
+    return data
+
+
 @_dask_cudf_nvtx_annotate
 def _get_non_empty_data(s):
-    if isinstance(s._column, cudf.core.column.CategoricalColumn):
+    if isinstance(s, cudf.core.column.CategoricalColumn):
         categories = (
-            s._column.categories
-            if len(s._column.categories)
-            else [UNKNOWN_CATEGORIES]
+            s.categories if len(s.categories) else [UNKNOWN_CATEGORIES]
         )
         codes = cudf.core.column.full(size=2, fill_value=0, dtype="int32")
-        ordered = s._column.ordered
+        ordered = s.ordered
         data = cudf.core.column.build_categorical_column(
             categories=categories, codes=codes, ordered=ordered
         )
+    elif isinstance(s, cudf.core.column.ListColumn):
+        leaf_type = s.dtype.leaf_type
+        if is_string_dtype(leaf_type):
+            data = ["cat", "dog"]
+        else:
+            data = np.array([0, 1], dtype=leaf_type).tolist()
+        data = _nest_list_data(data, s.dtype) * 2
```

Comment:
Does `s.dtype` need to be `s.dtype.leaf_type`? I'm not sure if this is nesting an extra level or if this is the right number of levels. It might have been changed with my previous round of suggestions.

Reply:
`leaf_type` corresponds to the terminal type of `ListDtype`, i.e., like `int32`, `int64`..

by passing `s.dtype`(`ListDtype`) we traverse through each inner type by accessing `.element_type` in `_nest_list_data` such that we get the true nesting levels.

Reply:
For example:
```python
>>> s.dtype
ListDtype(ListDtype(int64))
>>> s.leaf_type
int64
>>> s.element_type
ListDtype(int64)
```